compiler/tools/suggest.nim

```diff
@@ -211,12 +210,7 @@ proc `$`*(suggest: Suggest): string =
   result = $suggest.section
   result.add(sep)
   if suggest.section == ideHighlight:
-    if suggest.symkind.TSymKind == skVar and suggest.isGlobal:
```

Comment:
@alaviss has a good point, moving this logic into `nimsuggest` means that `suggest` becomes cleaner for other tools (lsp). The move will isolate the `nimsuggest` stuff so it won't break/regress anything in the meantime, and be a clean removal down the road.

Reply:
well, `$` is general proc , how to isolate it? maintain two versions of `$` ? and `nimlsp` also has `$` which I want to rename, as @alaviss said these values are safe to remove, can we just remove it?

Reply:
Basically nimsuggest usage of `$` is more like `toSuggestMsg` than `$` so yes you copy the old code over there.

After that you can change this one however you want.

Reply:
still `isGlobal` used in `symToSuggest`, as we can use a set of flag to mark as global, we don't need this.

Reply:
No, this isn't resolved. The functionality needs to move into nimsuggest so it doesn't regress. How that is determined, can happens via a flag.